File: Source Codes/Miscellaneous/Operations on Languages/language_operations.py
# Difference, Reverse, Kleene closure, Positive closure

# Concatenation function
def concat(set_1,set_2):
    z=set();
    for i in set_1:
        for j in set_2:
            z.add(i+j);
    return z;
    
set1=set();
set2=set();
len1=int(input("Enter number of elements of set1: "));
len2=int(input("Enter number of elements of set2: "));
print("\nEnter the elements of set1: ");
for i in range(len1):
    set1.add(input());
print("\nEnter the elements of set2: ");
for i in range(len2):
    set2.add(input());
print("\nSet 1:",set1);
print("Set 2:",set2);

# Difference
print("\nSet 1 - Set 2:",set1-set2);

# Reverse
rev=set();
for i in set1:
    rev.add(i [::-1]);
print("Reverse of set 1:",rev);

# Kleene closure
kleene={'lambda'};
temp=set();
count=int(input("\nEnter the count for kleene closure: "));
for i in range(1,count):
    if(i==1):
        temp=temp.union(set1);
    else:
        temp=concat(temp,set1);
    kleene=kleene.union(temp);
print("Kleene closure of set 1:",kleene);

# Positive closure
positive=set();
temp=set();
count=int(input("\nEnter the count for positive closure: "));
for i in range(count):
    if(i==0):
        temp=temp.union(set1);
    else:
        temp=concat(temp,set1);
    positive=positive.union(temp);
print("Positive closure of set 1:",positive);

File: Source Codes/Miscellaneous/Operations on Languages/language_operations.c
// Union, Intersection, Concatenation

#include<stdio.h>
#include<string.h>
int main()
{
    int len1,len2;
    printf("Enter the no of elements in set1 and set2\n");
    scanf("%d %d",&len1,&len2);
    char set1[len1+1][50],set2[len2+1][50];
    printf("Enter %d elements of set1\n",len1);
    for(int i=0;i<len1;i++)
        scanf("%s",set1[i]);
    printf("Enter %d elements of set2\n",len2);
    for(int i=0;i<len2;i++)
        scanf("%s",set2[i]);

    // Print set1
    printf("\nSet1\t\t: ");
    for(int i=0;i<len1;i++)
        printf("%s   ",set1[i]);

    // Print set2
    printf("\nSet2\t\t: ");
    for(int i=0;i<len2;i++)
        printf("%s   ",set2[i]);

    // Find Union
    printf("\nSet1 ⋃ Set2\t: ");
    for(int i=0;i<len1;i++)
        printf("%s   ",set1[i]);
    for(int i=0;i<len2;i++)
    {
        int flag=0;
        for(int j=0;j<len1;j++)
            if(!strcmp(set2[i],set1[j]))
            {
                flag=1;
                break;
            }
        if(flag==0)
            printf("%s   ",set2[i]);
    }

    // Find Intersection
    printf("\nSet1 ⋂ Set2\t: ");
    for(int i=0;i<len1;i++)
        for(int j=0;j<len2;j++)
            if(!strcmp(set1[i],set2[j]))
                printf("%s   ",set1[i]);

    // Find Concatenation
    printf("\nSet1 . Set2\t: ");
    for(int i=0;i<len1;i++)
        for(int j=0;j<len2;j++)
            printf("%s%s   ",set1[i],set2[j]);
}